web/src/main/java/com/epam/esm/web/controller/UserController.java

```diff
@@ -83,23 +78,32 @@ public UserController(UserService userService, PurchaseService purchaseService,
      *
      * @return list
      */
+    @PreAuthorize("isAuthenticated()")
     @GetMapping("/{id}/purchases")
-    public ResponseEntity<List<PurchaseDto>> findPurchaseByUserId(@PathVariable @Positive long id) throws RecourseNotExistException {
-        return new ResponseEntity(hateoasWrapper.hateoasWrapperListPurchaseDto(purchaseService.findListPurchaseByUserId(id)), HttpStatus.OK);
+    public ResponseEntity<List<PurchaseDto>> findPurchaseByUserId(
+            @PathVariable @Positive long id,
+            @Valid @RequestParam(required = false, value = PAGE_N, defaultValue = DEFAULT_PAGE_N)
+                    int pageN,
+            @Valid @RequestParam(required = false, value = PAGE_SIZE, defaultValue = DEFAULT_PAGE_SIZE)
+                    int pageSize) throws RecourseNotExistException {
+        return new ResponseEntity(hateoasWrapper.hateoasWrapperListPurchaseDto
+                (purchaseService.findListPurchaseByUserId(id, pageN, pageSize)), HttpStatus.OK);
     }
 
     /**
      * Create purchase for user
      *
      * @return PurchaseShortDto
      */
+    @PreAuthorize("hasAuthority('ROLE_USER')")
     @PostMapping("/{id}/purchases")
     public ResponseEntity<PurchaseShortDto> addPurchaseByUserId(@Valid @RequestBody PurchaseShortDto purchaseCreateDto,
```

Comment:
According to this, any user can add a purchase to any other user.
ONLY user should be authorized to acquire certificates for himself

Reply:
done